src/server/scripts/Zandalar/AtalDazar/boss_rezan.cpp

```diff
@@ -0,0 +1,350 @@
+    {
+        OnEffectHitTarget += SpellEffectFn(spell_rezan_tail_selector::HandleHitTarget, EFFECT_0, SPELL_EFFECT_DUMMY);
+    }
+};
+
+// 255421 - Devour
+class spell_rezan_devour : public AuraScript
+{
+    bool Validate(SpellInfo const* /*spellInfo*/) override
+    {
+        return ValidateSpellInfo({ SPELL_RIDE_VEHICLE });
+    }
+
+    void OnRemove(AuraEffect const* /*aurEff*/, AuraEffectHandleModes /*mode*/)
+    {
+        GetCaster()->RemoveAura(SPELL_RIDE_VEHICLE);
+    }
+
+    void Register() override
+    {
+        AfterEffectRemove += AuraEffectRemoveFn(spell_rezan_devour::OnRemove, EFFECT_1, SPELL_AURA_DUMMY, AURA_EFFECT_HANDLE_REAL);
+    }
+};
+
+// 257407 - Pursuit
+class spell_rezan_pursuit : public AuraScript
+{
+    bool Validate(SpellInfo const* /*spellInfo*/) override
+    {
+        return ValidateSpellInfo({ SPELL_REVERSE_CAST_RIDE_VEHICLE });
+    }
+
+    void HandlePeriodic(AuraEffect const* /*aurEff*/)
+    {
+        GetCaster()->CastSpell(GetTarget(), SPELL_REVERSE_CAST_RIDE_VEHICLE, true);
+    }
+
+    void Register() override
+    {
+        OnEffectPeriodic += AuraEffectPeriodicFn(spell_rezan_pursuit::HandlePeriodic, EFFECT_0, SPELL_AURA_PERIODIC_DUMMY);
+    }
+};
+
+// 255600 - Boss Emote Passed Points 0 @ Target
+class spell_rezan_boss_emote_at_target : public SpellScript
+{
+    void HandleHitTarget(SpellEffIndex /*effIndex*/)
+    {
+        if (Creature* casterCreature = GetCaster()->ToCreature())
+        {
+            if (!casterCreature)
+                return;
+
+            casterCreature->AI()->Talk(SAY_PURSUIT, GetHitUnit());
+        }
+    }
+
+    void Register() override
+    {
+        OnEffectHitTarget += SpellEffectFn(spell_rezan_boss_emote_at_target::HandleHitTarget, EFFECT_0, SPELL_EFFECT_DUMMY);
+    }
+};
+
+// 256608 - Pile of Bones
+struct at_rezan_pile_of_bones_spawn_raptor : AreaTriggerAI
+{
+    at_rezan_pile_of_bones_spawn_raptor(AreaTrigger* areatrigger) : AreaTriggerAI(areatrigger) { }
+
+    void OnUnitEnter(Unit* unit)
+    {
+        Creature* rezan = at->GetInstanceScript()->GetCreature(DATA_REZAN);
+        if (!rezan)
+            return;
+
+        if (rezan->IsInCombat())
+        {
+            if (!unit->IsPlayer() || (unit == rezan && !rezan->GetMap()->IsHeroicOrHigher()))
+                return;
+
+            rezan->CastSpell(at->GetPosition(), SPELL_PILE_OF_BONES_TRIGGER_SPAWN, true);
+            rezan->AI()->Talk(SAY_REANIMATED_RAPTOR_WARNING);
+            rezan->AI()->Talk(SAY_REANIMATED_RAPTOR_SUMMONER, unit);
+            at->Remove();
+        }
+    }
+};
+
+// 256606 - Pile of Bones
+struct at_rezan_pile_of_bones_slow : AreaTriggerAI
+{
+    at_rezan_pile_of_bones_slow(AreaTrigger* areatrigger) : AreaTriggerAI(areatrigger) { }
+
+    void OnUnitEnter(Unit* unit)
+    {
+        Creature* rezan = at->GetInstanceScript()->GetCreature(DATA_REZAN);
+        if (!rezan)
+            return;
+
+        if (rezan->IsInCombat())
```

Comment:
small adjustment, you can move this line up next to rezan and change the logic to avoid using {}

```
if (!rezan || !rezan->IsInCombat())
return;
```

Reply:
thanks